### PR TITLE
some comments on coding style

### PR DESCRIPTION
**Pull request by ThierryO**

### CODE REVIEW

File: gbif/gbif_name_match/gbif_species_name_match.R [hunk 1]
```diff
@@ -51,7 +51,9 @@ request_species_information <- function(df, name_col,
         do(as.data.frame(name_backbone(name = .[[name_col]]))) %>%
         select(gbif_terms) %>%
         bind_cols(df)
-    # (remark I use here Standard evaluation (SE) do instead of the NSE do_,
+    # (remark I use here Non Standard evaluation (SE) do instead of the SE do_,
```
Comment (stijnvanhoey) [inline gbif/gbif_name_match/gbif_species_name_match.R]: Indeed! Thanks for fixing this ;-)

**Comment by stijnvanhoey** [comment]
I've further adapted the `match.arg` to provide the usage of multiple terms and checking against the entire set of terms provided by the API. Can you have a quick look? I'm ok with merging these improvements.

**Comment by peterdesmet** [comment]
Looks fine as far as I can tell. Merge away!

File: gbif/gbif_name_match/gbif_species_name_match.R [hunk 2]
```diff
@@ -5,6 +5,12 @@ library('assertthat')
 library('assertable')
 
 
+API_terms = c('usageKey', 'scientificName', 'canonicalName', 'rank',
```
Comment (ThierryO) [inline gbif/gbif_name_match/gbif_species_name_match.R]: This is not good practice. Set this in the function arguments. That makes it more clear for the user. And will show up in the documentation (when adding to a package)

File: gbif/gbif_name_match/gbif_species_name_match.R [hunk 3]
```diff
@@ -44,7 +50,7 @@ request_species_information <- function(df, name_col,
     # test incoming arguments
     assert_that(is.data.frame(df))
     assert_colnames(df, name_col, only_colnames = FALSE) # colname exists in df
-    gbif_terms <- match.arg(gbif_terms)
+    gbif_terms <- match.arg(gbif_terms, API_terms, several.ok = TRUE)
```
Comment (ThierryO) [inline gbif/gbif_name_match/gbif_species_name_match.R]: `match.arg(gbif_terms, several.ok = TRUE)` after updating `gibf_terms` to the content of `API_terms`

File: gbif/gbif_name_match/gbif_species_name_match.R (hunk 2 above)
Comment (stijnvanhoey) [reply gbif/gbif_name_match/gbif_species_name_match.R]: I would rather not define the whole list in the argument, as we have a preferred set of defaults (the list as it is now). However, for checking if arguments make sense, the check against the entire list is required. 

Hence, providing it inside the function could be a possible consensus?

File: gbif/gbif_name_match/gbif_species_name_match.R (hunk 3 above)
Comment (stijnvanhoey) [reply gbif/gbif_name_match/gbif_species_name_match.R]: So, splitting this up in two checks? What would be the advantage?

Remark also that without the `several.ok = TRUE` addition, only the first value of the list is used, whereas we want to be able to add multiple of the terms at the same time (I'll clairfy this in docs).

**Comment by stijnvanhoey** [comment]
@ThierryO I've updated, but I'm not completely sure if I understand the 
`match.arg(gbif_terms, several.ok = TRUE) after updating gibf_terms to the content of API_terms` issue?

File: gbif/gbif_name_match/gbif_species_name_match.R (hunk 2 above)
Comment (ThierryO) [reply gbif/gbif_name_match/gbif_species_name_match.R]: In he typical use case of `match.arg()` all valid options are listed in the function argument and one the first one is selected by default.
It is now clear to me that you have a different use case. So `match.args(gbif_terms, api_terms)` is OK. However the documentation should be updated to reflect that.

Comment (stijnvanhoey) [reply gbif/gbif_name_match/gbif_species_name_match.R]: ok, thanks. Interesting functionality indeed. I've updated the docs as well.

Comment (ThierryO) [reply gbif/gbif_name_match/gbif_species_name_match.R]: Another cool thing is that `match.arg()` allows abbreviations.
```
test <- function(x = c("one", "two", "other")){
  x <- match.arg(x)
  cat(x)
}
test("t")
# two
test("o")
# Error in match.arg(x) : 'arg' should be one of “one”, “two”, “other”
test("ot")
# other
```

File: gbif/gbif_name_match/gbif_species_name_match.R [hunk 4]
```diff
@@ -47,6 +42,12 @@ request_species_information <- function(df, name_col,
                                                        'status',
                                                        'family')){
 
+    API_terms = c('usageKey', 'scientificName', 'canonicalName', 'rank',
```
Comment (ThierryO) [inline gbif/gbif_name_match/gbif_species_name_match.R]: me nitpicking: use `<-` for assignment. not `=`

Comment (stijnvanhoey) [reply gbif/gbif_name_match/gbif_species_name_match.R]: no worries, you're absolutely right

**Comment by stijnvanhoey** [comment]
Furhter improvements are always welcome, I'm merging here and now ;-) Thanks @DimEvil and @ThierryO

**Pull request merged**